e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js

```diff
@@ -65,6 +65,77 @@ describe("scenarios > embedding > dashboard parameters", () => {
   });
 
   context("UI", () => {
+    it("should allow searching dashboard parameters in preview embed modal", () => {
+      H.visitDashboard("@dashboardId");
+
+      cy.intercept(
+        "GET",
+        "api/preview_embed/dashboard/*",
+        cy.spy().as("previewEmbedSpy"),
+      ).as("previewEmbed");
+
+      H.openStaticEmbeddingModal({
+        activeTab: "parameters",
+        acceptTerms: true,
+        previewMode: "preview",
+      });
+
+      cy.wait("@previewEmbed", { timeout: 20000 });
+
+      cy.findByLabelText("Configuring parameters").as("allParameters");
+
+      cy.get("@allParameters").within(() => {
+        // select the dropdown next to the Name parameter so that we can set it to editable
+        cy.findByText("Name")
+          .parent()
+          .within(() => {
+            cy.findByText("Disabled").click();
+          });
+      });
+
+      H.popover().findByText("Editable").click();
+      cy.wait("@previewEmbed", { timeout: 20000 });
+
+      cy.get("iframe", { timeout: 30000 }).should("be.visible");
+
+      // Wait specifically for the Name parameter to be visible and editable in the iframe
+      H.getIframeBody().within(() => {
+        cy.findByTestId("dashboard-parameters-widget-container", {
+          timeout: 30000, // Longer timeout for throttled network
+        })
+          .should("be.visible")
+          .findByText("Name")
+          .should("be.visible");
+      });
+
+      cy.get("@previewEmbedSpy").should("have.callCount", 2);
+      H.getIframeBody().findByText("Test Dashboard").should("exist");
+
+      // Use the helper function which has built-in iframe loading logic
+      H.getIframeBody().within(() => {
+        // Wait for dashboard parameters to be available before interacting
+        cy.findByTestId("dashboard-parameters-widget-container", {
+          timeout: 20000,
```

Comment:
The longest test took about 30seconds to run. I don't understand why this test needs to override `timeout` in that many places. If you sum all the timeout added to this test it's 90 seconds which looks really excessive. I'd suggest you only put `timeout` on places where it's really needed otherwise, it's gonna be really hard to debug in the future if this test begin to flake.